### PR TITLE
Add uwp packaging logic

### PR DESCRIPTION
**Pull request by chrisfromwork**
This review adds uwp packaging logic to the pre-existing android/ios NPM package. It contains the following:

1. Modules\@babylonjs\react-native\windows\BabylonReactNative.sln has been removed. We will plan to build static libraries for the cmake generated BabylonNative solution (ReactBabylonNative.sln). We will build the BabylonReactNative.dll when it is autolinked into a react-native-windows project. Trying to prebuild the BabylonReactNative.dll seems like it will be prone to issues if folks have mismatching versions of react-native and react-native-windows.
1. Build.ps1 has been updated to build the cmake generated solution to generate static libs compared to building BabylonReactNative.sln
1. gulp tasks have been created to build and pack uwp resources. These tasks are defined independent to any preexisting android/iOS packaging tasks
1. PackageTest has been updated to support react-native-windows. It now also consumes generated uwp NPM packages.
1. The metro-config NPM package has been added back as a dev dependency for the Playground app. It seems that without this package declaration, metro is unable to resolve exclusionList. I believe in the past I may have had stale NPM packages allowing this to still work without declaring the devDependency

### CODE REVIEW

File: Apps/Playground/package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "@react-native-community/slider": "4.0.0-rc.2",
     "logkitty": "^0.7.1",
     "react": "^17.0.1",
-    "react-native": "^0.64.0-rc.0",
+    "react-native": "0.64.0-rc.0",
```
Comment (ryantrem) [inline Apps/Playground/package.json:21]: Why is this change needed?

File: Package/package.json [hunk 2]
```diff
@@ -3,6 +3,10 @@
   "devDependencies": {
     "fancy-log": "^1.3.3",
     "gulp": "^4.0.2",
+    "gulp-rename": "2.0.0",
     "shelljs": "^0.8.4"
+  },
+  "dependencies": {
+    "@babylonjs/react-native": "file:Assembled/babylonjs-react-native-0.0.1.tgz"
```
Comment (ryantrem) [inline Package/package.json:10]: Guessing this was inadvertent... revert?

**Comment by ryantrem** [comment]
Regarding this comment in your PR description:

> We will build the BabylonReactNative.dll when it is autolinked into a react-native-windows project.

Autolinking makes the `BabylonReactNative.vcxproj` get referenced as a dependency by the consuming RNW app, and the associated `BabylonReactNative.dll` is built when the consuming RNW app is built, correct?

File: Package/gulpfile.js [hunk 3]
```diff
@@ -93,6 +94,89 @@ const copyAndroidFiles = async () => {
...
+    .pipe(rename({ dirname: '' }))
+    .pipe(gulp.dest('Assembled/windows/libs/arm64/Release'));
+}
+
+const copyVCXProjUWPFiles = () => {
+  return gulp.src('../Modules/@babylonjs/react-native/windows/BabylonReactNative/*.*')
+    .pipe(gulp.dest('Assembled/windows/BabylonReactNative'));
+}
+
+const copyUWPFiles = gulp.series(
```
Comment (ryantrem) [inline Package/gulpfile.js:168]: Could this one actually be `gulp.parallel`?

File: Apps/PackageTest/metro.config.js [hunk 4]
```diff
@@ -4,13 +4,25 @@
...
+  resolver: {
+    blockList: exclusionList([
+      // This stops "react-native run-windows" from causing the metro server to crash if its already running
+      new RegExp(
+        `${path.resolve(__dirname, 'windows').replace(/[/\\]/g, '/')}.*`,
+      ),
+      // This prevents "react-native run-windows" from hitting: EBUSY: resource busy or locked, open msbuild.ProjectImports.zip
+      /.*\.ProjectImports\.zip/,
+    ]),
+  },
```
[the remark this answers is not in the document]
Comment (chrisfromwork) [reply Apps/PackageTest/metro.config.js:20]: react-native-windows on setup overwrites this file to contain this logic, and when other custom things are in this file it doesn't seem to merge them. If there are thoughts on how things could be improved, we should relay this feedback to the react-native-windows team.

File: Apps/Playground/package.json (hunk 1 above)
Comment (chrisfromwork) [reply Apps/Playground/package.json:21]: when running react-native-windows initialization, react-native-windows will prompt/force users to use newer preview releases that support whatever react-native release is in the project. I don't think we are guaranteed that react-native-windows preview releases won't contain breaking changes (I have seen breaking changes already).

What I was seeing was that react-native would get installed at 0.64.0-rc.2, forcing us to update to react-native-windows 0.64.0-preview.7 which contained breaking changes for native modules (This was all during initialization prompts compared to an action of npm package installs). I believe we can ensure things will compile if we don't auto update to newer versions of react-native/react-native-windows. I think this will probably get better when react-native-windows has a formal 0.64.0 release.

File: Package/gulpfile.js (hunk 3 above)
Comment (chrisfromwork) [reply Package/gulpfile.js:168]: yeah seems like it.

File: Apps/PackageTest/metro.config.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (ryantrem) [reply Apps/PackageTest/metro.config.js:20]: Yea that's pretty bad. I guess their assumption is that people will diff the file and manually fix it up. One thing that would help would be to put this logic in a separate file that is bundled with RNW and import it into this file. That might be good feedback for the rnw folks.

**Review by ryantrem — APPROVED** [review]
